src/test/java/br/com/exemplo/springboot/service/ClientServiceTest.java

```diff
@@ -29,55 +37,203 @@ public void beforeMethod(){
+        Client client1 = new Client(1L, "vinicius", Gender.MALE);
+        Client client2 = new Client(2L, "Maria", Gender.FEMALE);
+        Client client3 = new Client(3L, "Maria", Gender.FEMALE);
+        clients.add(client1);
+        clients.add(client2);
+        clients.add(client3);
+
+        when(clientRepository.findByName("Maria")).thenReturn(clients.stream()
+                .filter(x -> x.getName().equals("Maria")).collect(Collectors.toList()));
+        List<ClientDto> clientsDto = this.clientService.getByName("Maria");
+        verify(clientRepository).findByName("Maria");
+
+        assertEquals(clientsDto.size(), 2);
+    }
+
+    @Test(expectedExceptions = NullPointerException.class)
+    public void getByNameTest_FindNull() {
+        ReflectionTestUtils.setField(clientService, "modelMapper", new ModelMapper());
+
+        List<Client> clients = new ArrayList<>();
+        Client client1 = new Client(1L, "vinicius", Gender.MALE);
+        Client client2 = new Client(2L, "Maria", Gender.FEMALE);
+        Client client3 = new Client(3L, "Maria", Gender.FEMALE);
+        clients.add(client1);
+        clients.add(client2);
+        clients.add(client3);
+
+        when(clientRepository.findByName("Bob")).thenReturn(clients.stream()
+                .filter(x -> x.getName().equals("Bob")).collect(Collectors.toList()));
+        List<ClientDto> clientsDto = this.clientService.getByName(null);
+        verify(clientRepository).findByName("Bob");
+
+        assertEquals(clientsDto.size(), 0);
     }
 
     @Test
-    public void saveTest() {
+    public void saveTest() throws ParseException {
+        ReflectionTestUtils.setField(clientService, "modelMapper", new ModelMapper());
+
+        ClientDto clientDto = new ClientDto(null, "vinicius", Gender.MALE);
+        Client client = new Client(1L, "vinicius", Gender.MALE);
+
+        when(clientRepository.save(any())).thenReturn(client);
+        ClientDto clientsDto2 = this.clientService.save(clientDto);
+        verify(clientRepository).save(any());
+
+        clientDto.setId(clientsDto2.getId());
+        assertEquals(clientDto,clientsDto2);
+    }
+
+    @Test(expectedExceptions = NullPointerException.class)
+    public void saveTest_NullClient() throws ParseException {
+        ReflectionTestUtils.setField(clientService, "modelMapper", new ModelMapper());
 
+        when(clientRepository.save(any())).thenReturn(null);
+        ClientDto clientsDto2 = this.clientService.save(null);
+        verify(clientRepository).save(any());
+
+        assertEquals(clientsDto2, null);
     }
 
     @Test
-    public void updateTest() {
+    public void updateTest() throws ParseException {
+        ReflectionTestUtils.setField(clientService, "modelMapper", new ModelMapper());
+
+        ClientDto clientDto = new ClientDto(1L, "vinicius", Gender.MALE);
+        Client client = new Client(1L, "vinicius", Gender.MALE);
+
+        when(clientRepository.save(any())).thenReturn(client);
```

Comment:
Evitar usar o any() genérico.
```suggestion
        when(clientRepository.save(any(ClientDto.class))).thenReturn(client);
```